wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/CommandConverter.java

```diff
@@ -94,9 +100,32 @@ public CommandImpl toDevfileCommand(
       command.getAttributes().put(WORKING_DIRECTORY_ATTRIBUTE, commandAction.getWorkdir());
     }
 
-    command
-        .getAttributes()
-        .put(Constants.COMPONENT_ALIAS_COMMAND_ATTRIBUTE, commandAction.getComponent());
+    if (commandAction.getComponent() != null) {
+      command
+          .getAttributes()
+          .put(Constants.COMPONENT_ALIAS_COMMAND_ATTRIBUTE, commandAction.getComponent());
+    }
+
+    if (commandAction.getReference() != null) {
+      command.getAttributes().put(COMMAND_ACTION_REFERENCE_ATTRIBUTE, commandAction.getReference());
+    }
+
+    if (commandAction.getReferenceContent() != null) {
+      command
+          .getAttributes()
+          .put(COMMAND_ACTION_REFERENCE_CONTENT_ATTRIBUTE, commandAction.getReferenceContent());
+    } else if (commandAction.getReference() != null) {
+      try {
+        String referenceContent = contentProvider.fetchContent(commandAction.getReference());
+        command.getAttributes().put(COMMAND_ACTION_REFERENCE_CONTENT_ATTRIBUTE, referenceContent);
+      } catch (IOException e) {
+        throw new DevfileException(
+            format(
+                "Failed to fetch content of action from reference %s",
```

Comment:
I think it's better to add a cause to the error message, otherwise, end-user won't be able to see an original error, and it may be like `404 Page not found`, or `invalid URL`. WDYT?

Reply:
addressed.